int8 tid
string trajectory_type
